src/Database/FieldTypeConverter.php

```diff
@@ -124,20 +74,23 @@ public function __construct(TypeMap $typeMap, DriverInterface $driver)
      */
     public function __invoke(mixed $row): mixed
     {
-        if (!is_array($row)) {
+        if (!is_array($row) || !count($row)) {
             return $row;
         }
 
-        if (!empty($this->_typeMap)) {
-            foreach ($this->_typeMap as $field => $type) {
-                $row[$field] = $type->toPHP($row[$field], $this->_driver);
+        foreach ($this->typeFields as $typeName => $fields) {
+            $type = $this->types[$typeName];
+            if ($type instanceof OptionalConvertInterface && !$type->requiresToPhpCast()) {
```

Comment:
Doesn't doing these checks on each field conversion add overhead to the 'hot' path of this class?

Reply:
I am not sure this interface is useful anymore. With PDO, you can get a typed results back such as an integer and this would prevent casting it to a string in cake.

Reply:
Moved all checks into the constructor.